### DISCUSSION TITLE
AVIF Support

### DISCUSSION THREAD

**Discussion by Nuranto** [spatie/laravel-medialibrary#2664] [Ideas]
Would be great to have AVIF support !

see https://www.php.net/manual/en/function.imagecreatefromavif.php

It's only available in PHP8 though...

Also : https://github.com/spatie/image/releases/tag/2.2.0

**Comment by findgr** [comment]
There is time, until all browsers add full support.

**Reply by Nuranto** [reply]
https://caniuse.com/?search=avif

Only edge doesnt not support it at the moment. Also, there will always be old browser that won't support it, as for webp. 
That doesn't mean we cannot use it right now (Edge is only 4% of the market, and multi-source images can be used to deal with him.)

**Reply by findgr** [reply]
Safari has partial support also

**Comment by MACscr** [comment]
We need avif support as well. The GD library supports it and we run php 8.2.

**Comment by francoism90** [comment]
So this has been merged: https://github.com/spatie/laravel-medialibrary/commit/83e902ed0d4204a217e19e7a764b33236c1d4526

@0xb4lint @freekmurze are the following changes correct:
```php
    public function convert(string $file, Conversion $conversion = null): string
    {
        $pathToImageFile = pathinfo($file, PATHINFO_DIRNAME).'/'.pathinfo($file, PATHINFO_FILENAME).'.png';

        $image = imagecreatefromavif($file);

        imagepng($image, $pathToImageFile, 9);

        imagedestroy($image);

        return $pathToImageFile;
    }
```
Why is a PNG created and is it correct `$pathToImageFile` has the `.png` extension? Shouldn't it be `.avif`?

Thanks.

**Reply by 0xb4lint** [reply]
The `Spatie\MediaLibrary\Conversions\ImageGenerators\Avif` is for creating png file from avif.

**Reply by francoism90** [reply]
@0xb4lint Ah, sorry. :)
